ipynb/tests/Frequency_Invariance_Test.ipynb

```diff
@@ -0,0 +1,799 @@
+    "\n",
+    "The test class runs the same workload at a selection of frequencies, each entry in `t.experiments` represents a run at a different frequency."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import json\n",
+    "\n",
+    "from tests.eas.load_tracking import FreqInvarianceTest\n",
+    "from trace import Trace\n",
+    "from trappy.plotter import plot_trace\n",
+    "from trappy.stats.grammar import Parser\n",
+    "from trappy import ILinePlot"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2016-12-13 18:42:57,482 INFO    : root         : Using LISA logging configuration:\n",
+      "2016-12-13 18:42:57,483 INFO    : root         :   /home/brejac01/sources/lisa/logging.conf\n"
+     ]
+    }
+   ],
+   "source": [
+    "import logging\n",
+    "from conf import LisaLogging\n",
+    "LisaLogging.setup()\n",
+    "logging.getLogger('Analysis').setLevel(logging.ERROR)\n",
+    "logging.getLogger('Trace').setLevel(logging.ERROR)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run test workload\n",
+    "\n",
+    "There's currently no way to pass a `TestEnv` or configuration to automated test classes. Instead the target information comes from the `target.config` file (in the root of the LISA source tree), so you'll need to edit that to configure LISA to connect to your target."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "    Frequency invariance test for util_avg signal\n",
+      "\n",
+      "    Run workloads providing a known utilization and test that the PELT\n",
+      "    (Per-entity load tracking) util_avg and load_avg signals reflects that\n",
+      "    utilization at various CPU frequencies.\n",
+      "    \n"
+     ]
+    }
+   ],
+   "source": [
+    "t = FreqInvarianceTest()\n",
+    "print t.__doc__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To run automated tests from within a notebook we instantiate the test class and call `setUpClass` on it. This is what `nosetests` does internally when running tests from the command line."
```

Comment:
Yeah, that sounds like a good idea. The whole way we use `unittest` is a bit iffy IMO but maybe at least we can hide it. I'll see what I can do.